### PR TITLE
Feat: Added pollingTimeout and blockingTimeoutPeriod setters in optimizelyFactory

### PR DESCRIPTION
**Pull request by mnoman09**

### CODE REVIEW

File: OptimizelySDK/OptimizelyFactory.cs [hunk 1]
```diff
@@ -35,10 +35,22 @@ public static class OptimizelyFactory
     {
         private static int MaxEventBatchSize;
         private static TimeSpan MaxEventFlushInterval;
+        private static TimeSpan PollingInterval;
+        private static TimeSpan BlockingTimeoutPeriod;
         private static ILogger OptimizelyLogger;
         private const string ConfigSectionName = "optlySDKConfigSection";
 
 #if !NETSTANDARD1_6 && !NET35
+        public static void SetPollingInterval(TimeSpan pollingInterval)
```
Comment (msohailhussain) [inline OptimizelySDK/OptimizelyFactory.cs]: please add comments

File: OptimizelySDK/OptimizelyFactory.cs [hunk 2]
```diff
@@ -35,10 +35,22 @@ public static class OptimizelyFactory
...
+        {
+            PollingInterval = pollingInterval;
+        }
+
+        public static void SetBlockingTimeoutPeriod(TimeSpan blockingTimeoutPeriod)
```
Comment (msohailhussain) [inline OptimizelySDK/OptimizelyFactory.cs]: same here.

**Comment by msohailhussain** [comment]
add unit test for these changes, also can you please mention what's the priority if I use settings from app.config and provide pollinginterval in the settings as well.

**Comment by mikeng13** [comment]
Also, please fix the merge conflicts!

**Comment by mikeng13** [comment]
@mnoman09 please fix merge conflicts!

File: OptimizelySDK.Tests/App.config [hunk 3]
```diff
@@ -12,10 +12,10 @@
   
   <optlySDKConfigSection>
   
-    <HttpProjectConfig sdkKey="43214321" 
+    <HttpProjectConfig sdkKey="QBw9gFM8oTn7ogY9ANCC1z"
```
Comment (msohailhussain) [inline OptimizelySDK.Tests/App.config]: Please use any dummy datafile key.

**Pull request closed without merging**